[19.0][MIG] attachment_delete_restrict: Migration to 19.0

AungKoKoLin1997 opened the pull request:
- Remove the self.env.su condition in unlink, since it is always called with sudo() in Odoo 19.
https://github.com/odoo/odoo/blob/98bae80f57733162c67593539ef47dfd0c6d8797/addons/mail/controllers/attachment.py#L98

@qrtl QT5667

yostashiro committed:
[ADD] attachment_delete_restrict

Kev-Roche committed:
[IMP] attachment_delete_restrict: black, isort, prettier

Kev-Roche committed:
[MIG] attachment_delete_restrict: Migration to  14.0

Kev-Roche committed:
rework module attachment_delete_restrict (to squash after review)

sebastienbeau committed:
attachment_delete_restrict: refactor test and improve restrict mode naming

sebastienbeau committed:
attachment_delete_restrict: refactor code

Commit:
[UPD] Update attachment_delete_restrict.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] icon.png

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (37 of 37 strings)

Translation: server-tools-14.0/server-tools-14.0-attachment_delete_restrict
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-attachment_delete_restrict/es_AR/

Commit:
[FIX] attachment_delete_restrict: adapt to Odoo permission change

OCA-git-bot committed:
attachment_delete_restrict 14.0.1.0.1

AungKoKoLin1997 committed:
[MIG] attachment_delete_restrict: Migration to 16.0

AungKoKoLin1997 committed:
[FIX] attachment_delete_restrict
- Remove the display of duplicate allowed user names.
- Correct the many2many field relation
- Correct domain values and add the many2many_tags widget in views.

AungKoKoLin1997 committed:
[IMP] attachment_delete_restrict
- Add _onchange_restrict_delete_attachment() to clear groups and users when restrict_delete_attachment is not set to custom or owner_custom.

sebastienbeau committed:
attachment_delete_restrict: using sudo should by pass the restriction

Commit:
[UPD] Update attachment_delete_restrict.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (38 of 38 strings)

Translation: server-tools-16.0/server-tools-16.0-attachment_delete_restrict
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-attachment_delete_restrict/it/

AungKoKoLin1997 committed:
[IMP] attachment_delete_restrict: pre-commit auto fixes

AungKoKoLin1997 committed:
[MIG] attachment_delete_restrict: Migration to 18.0

AungKoKoLin1997 committed:
[IMP] attachment_delete_restrict: imp readme

AungKoKoLin1997 committed:
[IMP] attachment_delete_restrict: pre-commit auto fixes

AungKoKoLin1997 committed:
[MIG] attachment_delete_restrict: Migration to 19.0

- Remove the self.env.su condition in unlink, since it is always called with sudo() in Odoo 19.
https://github.com/odoo/odoo/blob/98bae80f57733162c67593539ef47dfd0c6d8797/addons/mail/controllers/attachment.py#L98

AungKoKoLin1997 committed:
[MIG] attachment_delete_restrict: Migration to 19.0

- Remove the self.env.su condition in unlink, since it is always called with sudo() in Odoo 19.
https://github.com/odoo/odoo/blob/98bae80f57733162c67593539ef47dfd0c6d8797/addons/mail/controllers/attachment.py#L98

yostashiro commented on attachment_delete_restrict/models/ir_attachment.py:
```diff
@@ -0,0 +1,82 @@
+            return self._raise_delete_attachment_error(allowed_users)
+
+    def unlink(self):
+        if self.env.uid == SUPERUSER_ID:
```
I thought there was a possibility that this change might cause inconveniences in some operational scenarios.

@sebastienbeau Do you have any concerns on this? If you do, we might revert this change and additionally insert a check at the beginning of `mail_attachment_delete()`.

AungKoKoLin1997 committed:
[FIX] attachment_delete_restrict: avoid KeyError on orphan res_model

When an attachment's res_model no longer resolves to a registered ir.model
(e.g. left over from an uninstalled module), unlink raised a KeyError. Use
dict.get() and fall back to the global configuration when the model is not
found.

AungKoKoLin1997 committed:
[FIX] attachment_delete_restrict: resolve implied group membership

In Odoo 19 res.groups.user_ids holds only the users explicitly added to
the group; users that belong through group implication are exposed via
the implication-aware all_user_ids field (has_group uses the same
semantics). The migration mechanically renamed the former .users to
.user_ids, which silently narrowed the deletion authorization: a user
holding an authorized deletion group only via implication was wrongly
denied. Use all_user_ids when expanding the allowed users, and add a
regression test covering the implied-group case.
